core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -93,6 +94,14 @@ public int compare(RecordPointerAndKeyPrefix r1, RecordPointerAndKeyPrefix r2) {
   private int pos = 0;
 
   /**
+   * If sorting with radix sort, specifies the starting position in the sort buffer where records
+   * with non-null prefixes are kept. Positions [0..nullPos) will contain null-prefixed records,
+   * and positions [nullPos..pos) non-null prefixed records. This lets us avoid radix sorting
+   * over null values.
+   */
+  private int nullPos = 0;
```

Comment:
minor nit: would something `nullBoundary` or `nullBoundaryPos` better imply the semantics?